src/handlers/TechRecordsListHandler.ts

```diff
@@ -0,0 +1,165 @@
+import TechRecordsDAO from "../models/TechRecordsDAO";
+import { ISearchCriteria } from "../../@Types/ISearchCriteria";
+import { HTTPRESPONSE, SEARCHCRITERIA, STATUS } from "../assets/Enums";
+import HTTPError from "../models/HTTPError";
+import { cloneDeep } from "lodash";
+import { Vehicle } from "../../@Types/TechRecords";
+import { ErrorHandler } from "./ErrorHandler";
+
+export class TechRecordsListHandler<T extends Vehicle> {
+  private readonly techRecordsDAO: TechRecordsDAO;
+
+  constructor(techRecordsDAO: TechRecordsDAO) {
+    this.techRecordsDAO = techRecordsDAO;
+  }
+
+  /* #region  Public functions */
+  public async getFormattedTechRecordsList(
+    searchTerm: string,
+    status: string,
+    searchCriteria: ISearchCriteria = SEARCHCRITERIA.ALL
+  ): Promise<T[]> {
+    try {
+      // Formatting the object for lambda function
+      let techRecordItems = await this.getTechRecordList(
+        searchTerm,
+        status,
+        searchCriteria
+      );
+      techRecordItems = this.formatTechRecordItemsForResponse(techRecordItems);
+
+      return techRecordItems;
+    } catch (error) {
+      if (!(error instanceof HTTPError)) {
+        console.error(error);
+        error.statusCode = 500;
+        error.body = HTTPRESPONSE.INTERNAL_SERVER_ERROR;
+      }
+      throw ErrorHandler.Error(error.statusCode, error.body);
+    }
+  }
+
+  public async getTechRecordList(
+    searchTerm: string,
+    status: string,
+    searchCriteria: ISearchCriteria = SEARCHCRITERIA.ALL
+  ): Promise<T[]> {
+    const data: any = await this.techRecordsDAO.getBySearchTerm(
+      searchTerm,
+      searchCriteria
+    );
+    if (data.Count === 0) {
+      throw new HTTPError(404, HTTPRESPONSE.RESOURCE_NOT_FOUND);
+    }
+    // Formatting the object for lambda function
+    let techRecordItems: T[] = data.Items;
+    if (status !== STATUS.ALL) {
+      techRecordItems = this.filterTechRecordsByStatus(techRecordItems, status);
+    }
+    return techRecordItems;
+  }
+
+  public formatTechRecordItemForResponse(techRecordItem: T) {
+    // Adding primary and secondary VRMs in the same array
+    const vrms = [];
+    if (techRecordItem.primaryVrm) {
+      vrms.push({ vrm: techRecordItem.primaryVrm, isPrimary: true });
+    }
+    if (techRecordItem.secondaryVrms) {
+      for (const secondaryVrm of techRecordItem.secondaryVrms) {
+        vrms.push({ vrm: secondaryVrm, isPrimary: false });
+      }
+    }
+    Object.assign(techRecordItem, {
+      vrms
+    });
+    // Cleaning up unneeded properties
+    delete techRecordItem.primaryVrm; // No longer needed
+    delete techRecordItem.secondaryVrms; // No longer needed
+    delete techRecordItem.partialVin; // No longer needed
+
+    techRecordItem.techRecord.forEach((techRecord: any) => {
```

Comment:
We could be using a filter here I think to make it cleaner and return it ?
return techRecord
  .filter(({euroStandard}) => euroStandard !== null && ...)
 .map(({euro}) => euro.toString())

Reply:
The filter returns a new array of filtered elements and the map updates the values for the filtered array. The foreach updates the values on the same array.

Reply:
I leave it to you, I had in mind returning the new array which then would be piped as input into the next function.
I thought it would be more concise and shorter